Three tests fail on Windows: TOML escaping, POSIX paths, wiped env

Description: Three independent tests in the suite make Windows-incompatible assumptions, all failing when run on a Windows machine even though the code under test is correct. (1) `tests/evals/core/test_config_split_parsing.py:18` interpolates a raw Windows `tmp_path` (e.g. `C:\Users\...\out`) directly inside a TOML basic (double-quoted) string; TOML basic strings treat backslash as an escape introducer, so a segment like `\Users` parses as the invalid escape `\U`, and `tomllib` rejects it. (2) `tests/learning/test_trial_runner.py:140` asserts `cfg.output_path.startswith("out/")`, a POSIX-only literal, against a path built with `Path(self.output_dir) / f"..."`, which legitimately produces backslashes on Windows. (3) `tests/evals/comparison/test_openclaw_runner_contract.py:38-41` builds `env = {"OPENCLAW_PATH": ..., "HOME": ...}` and passes it as the entire environment to `subprocess.run(..., env=env)`, wiping `TEMP`/`TMP`/`SystemRoot`/`PATH`; Node's `os.tmpdir()` (used by the runner script under test) then falls back to the literal string `'undefined'` when none of those are set, producing an unusable temp path and an `ENOENT` from `mkdtempSync`.

```python
# test_config_split_parsing.py:18
output_dir = "{tmp_path / "out"}"
```
```python
# test_trial_runner.py:140
assert cfg.output_path.startswith("out/")
```
```python
# test_openclaw_runner_contract.py:36-41
env = {
    "OPENCLAW_PATH": str(tmp_path),
    "HOME": str(tmp_path / "home"),
}
result = subprocess.run([node, str(runner), ...], env=env, ...)
```

Steps to Reproduce (on Windows):
1. `pytest tests/evals/core/test_config_split_parsing.py` - observe all 5 tests fail with `tomllib.TOMLDecodeError: Invalid hex value (at line 6, column 19)`.
2. `pytest tests/learning/test_trial_runner.py -k test_output_path_contains_trial_id` - observe `AssertionError: assert False where False = 'out\\trial-abc_qwen3-8b.jsonl'.startswith('out/')`.
3. `pytest tests/evals/comparison/test_openclaw_runner_contract.py::test_openclaw_runner_parses_real_agent_json_shape` (requires `node` on PATH) - observe the assertion on `result.returncode == 0` fails with a Node `ENOENT`/`mkdtemp` stack trace.

Expected Behavior: The full test suite passes on Windows; none of these three assertions/fixtures should depend on POSIX-only path conventions or on the parent process's environment being irrelevant to the subprocess under test.

Actual Behavior: All three fail on Windows, for three distinct platform-specific reasons.

Environment: OpenJarvis @ 1ca3ec6 (main), Python 3.13.1, Windows 11, Node.js present on PATH. Windows-only - these tests pass as written on POSIX systems.

Suggested fix: (1) Render the TOML path with `.as_posix()` so it is unambiguous inside a TOML basic string on both platforms. (2) Replace the POSIX-literal `startswith("out/")` check with an `os.sep`-agnostic comparison, e.g. `Path(cfg.output_path).parts[0] == "out"`. (3) Build the child environment as `{**os.environ, "OPENCLAW_PATH": ..., "HOME": ..., "USERPROFILE": ...}` so the parent's `TEMP`/`TMP`/`SystemRoot`/`PATH` are inherited while still overriding the test-specific variables. A local patch making all three changes is available if useful.

--- a/tests/evals/comparison/test_openclaw_runner_contract.py
+++ b/tests/evals/comparison/test_openclaw_runner_contract.py
@@ -3,6 +3,7 @@
 from __future__ import annotations
 
 import json
+import os
 import shutil
 import subprocess
 from pathlib import Path
@@ -35,9 +36,17 @@ def test_openclaw_runner_parses_real_agent_json_shape(tmp_path: Path) -> None:
         / "src/openjarvis/evals/backends/external/_runners/openclaw_runner.mjs"
     )
     out_json = tmp_path / "out.json"
+    # subprocess.run's `env` REPLACES the environment rather than merging
+    # it, so passing just these two keys wiped everything Node needs to
+    # even start on Windows (TEMP/TMP/SystemRoot/PATH) -- os.tmpdir()
+    # falls back to the literal string 'undefined' when those are unset,
+    # and mkdtemp then fails with ENOENT (#785). Inherit the parent
+    # environment and layer the test-specific overrides on top.
     env = {
+        **os.environ,
         "OPENCLAW_PATH": str(tmp_path),
         "HOME": str(tmp_path / "home"),
+        "USERPROFILE": str(tmp_path / "home"),
     }
     result = subprocess.run(
         [

--- a/tests/evals/core/test_config_split_parsing.py
+++ b/tests/evals/core/test_config_split_parsing.py
@@ -15,7 +15,7 @@ def _write_config(tmp_path: Path, split_value: str | None) -> Path:
 name = "split-parse-test"
 
 [run]
-output_dir = "{tmp_path / "out"}"
+output_dir = "{(tmp_path / "out").as_posix()}"
 seed = 42
 
 [[models]]

--- a/tests/learning/test_trial_runner.py
+++ b/tests/learning/test_trial_runner.py
@@ -2,6 +2,7 @@
 
 from __future__ import annotations
 
+from pathlib import Path
 from unittest.mock import MagicMock, patch
 
 from openjarvis.evals.core.types import RunConfig, RunSummary
@@ -137,7 +138,9 @@ class TestBuildRunConfig:
         cfg = runner._build_run_config(trial, recipe)
 
         assert "trial-abc" in cfg.output_path
-        assert cfg.output_path.startswith("out/")
+        # Path(...) / "..." legitimately produces backslashes on Windows,
+        # so compare path components rather than a hard-coded "out/".
+        assert Path(cfg.output_path).parts[0] == "out"
 
     def test_default_model_fallback(self) -> None:
         runner = TrialRunner(benchmark="supergpqa")

